docs/src/guide/parallelization.md

```diff
@@ -52,60 +52,109 @@ as a breakdown over individual routines.
+such that it cannot be used for calculations with only a single `k`-Point.
+Otherwise combining both forms of parallelism is possible as well.
+
+The scaling of both forms of parallelism for a number of test cases
+is demonstrated in the following figure.
+These values were obtained using DFTK version 0.1.17 and Julia 1.6
+and the precise scalings will likely be different
+depending on architecture, DFTK or Julia version.
+The rough trends should, however, be similar.
+
+```@raw html
+<img src="scaling.png" width=750 />
 ```
-using FFTW
-FFTW.set_num_threads(N)
+
+The MPI-based parallelisation strategy clearly shows a superior scaling
+and should be preferred if available.
+
+
+## MPI-based parallelism
+Currently DFTK uses MPI to distribute on `k`-Points only.
+This implies that calculations with only a single `k`-Point
+cannot use make use of this.
+The recommended way to use MPI with DFTK is to
+disable all threading and run Julia with the number
+of desired MPI processes.
+For details on setting up and configuring MPI with Julia
+see the [MPI.jl documentation](https://juliaparallel.github.io/MPI.jl/stable/configuration).
+
+1. To ensure the thread setup in DFTK is consistent with the environment
+   under which DFTK runs
+   add the following to your script running the DFTK calculation:
+   ```julia
+   using DFTK
+   setup_threading()
+   ```
+   This will ensure that all threading is disabled
+   (which unfortunately is not the default).
+
+2. Run Julia in parallel using the `mpiexecjl` wrapper script from MPI.jl:
+   ```sh
+   mpiexecjl -np 16 julia -t 1 myscript.jl
+   ```
+   In this `-np 16` tells MPI to use 16 processes and `-t 1` tells Julia
+   to use one thread only.  
+   Notice that we use `mpiexecjl` to automatically select the `mpiexec`
+   compatible with the MPI version used by Julia.
+
+As usual with MPI printing will be garbled. You can use
+```julia
+DFTK.mpi_master() || (redirect_stdout(); redirect_stderr())
 ```
-where `N` is the number of threads you desire.
+at the top of your script to disable printing on all processes but one.
 
+!!! warning "MPI-based parallelism is experimental"
+    Even though MPI-based parallelism shows the better scaling it is still
+    experimental and some routines (e.g. band structure and direct minimization)
+    are not compatible with it yet.
+
+## Thread-based parallelism
+Threading in DFTK currently happens on multiple layers
+distributing the workload
+over different ``k``-Points, bands or within
+an FFT or BLAS call between processors.
+At its current stage our scaling for thread-based parallelism
+is worse compared MPI-based and therefore the parallelism
+described here should
+only be used if no other option exists.
+To use thread-based parallelism proceed as follows:
+
+1. Ensure that threading is properly setup inside DFTK by adding
+   to the script running the DFTK calculation:
+   ```julia
+   using DFTK
+   setup_threading()
+   ```
```

Comment:
which disables FFT threading and sets the number of BLAS threads to the number of Julia threads